### ISSUE TITLE
Misplaced Glyphs - Font Fallback?

### ISSUE THREAD

**Issue by aphirst** [libass/libass#170]
Sample file: http://a.pomf.se/cnytpe.mkv

**Expected output:** http://a.pomf.se/mxiouq.png
**What I get instead:** http://a.pomf.se/bzurgg.jpg
Notice that the emoticon at ~`00:11` is garbled and ultimately spills outside of the screen.

@rcombs and `torque` said on IRC that this was probably a font fallback issue:
```
<torque> probably a fallback issue because the font they're using doesn't have those glyphs?
<torque> or harfbuzz causing problems
<rcombs> http://puu.sh/gZBFU/69c7623ead.png even better here
<rcombs> fontconfig: cannot find glyph U+FF08 in font 'Consolas', falling back to 'SimSun'
<rcombs> yeah, looking like a fallback issue
<rcombs> remind me what xy-vs does for that? "always fallback on Arial"?
<rcombs> there's definitely some fallback awkwardness going on, though like torque said harfbuzz could also be causing oddities
```

On my setup, the font fallback happens as follows:
```
[libass] fontconfig: cannot find glyph U+FF08 in font 'Consolas', falling back to 'Noto Sans CJK JP Regular'
```

As ever, if there's any other information I ought to provide, then just ask. :smile: 

**My system:**
* Arch Linux x86_64
* libass `0.12.1`
* mpv `0.8.3`
* freetype2-infinality-ultimate `2.5.5`
* fontconfig-infinality-ultimate `2.11.1`
* harfbuzz `0.9.40`

**Comment by lachs0r** [comment]
This looks okay-ish with fallback to MS PGothic (happens on my system).

**Comment by robin007bond** [comment]
I can confirm this problem. My fallback is:
`fontconfig: cannot find glyph U+FF08 in font 'Consolas', falling back to 'IPAGothic'

It looks worse than yours, haha:

http://a.pomf.se/ynfqpn.jpg

**Comment by grigorig** [comment]
I'm not sure what we are supposed to do in these cases. Obviously we can't do the same fallbacks as Windows without all the fonts installed. I'm considering a "prefer Arial as first fallback" thing for the fonts branch, but this wouldn't help here either. Closing.

**Comment by aphirst** [comment]
I decided to come back to this just for a laugh. When I run the file now (which thankfully is preserved on the web archive, I'll upload it elsewhere and edit my post later) I get the following fallback replacement:
```
[sub/ass] fontselect: (Consolas, 400, 0) -> Consolas, 0, Consolas
[sub/ass] Glyph 0xFF08 not found, selecting one more font for (Consolas, 400, 0)
[sub/ass] fontselect: (Consolas, 400, 0) -> /usr/share/fonts/TTF/DroidSansFallback.ttf, 0, DroidSansFallback
```
![mpv-shot0039](https://user-images.githubusercontent.com/1735556/28995658-d44fdb52-79e6-11e7-8298-1e72f5ca8e74.jpg)
Which looks even worse. I tried installing `inconsolita` to no avail, and also `otf-overpass`, which respectively seemed to provide consolas and gothic compatible fonts (though that may have been misunderstanding on my part).

I wonder whether there's any non-MS font which does look better with this, and if so, how to ensure that the fallback actually selects it.
